### PR TITLE
Implemented user tastes DB and repository

### PR DESCRIPTION
**Pull request by klesogor**

### CODE REVIEW

File: app/Repositories/User/UserRepository.php [hunk 1]
```diff
@@ -45,4 +46,24 @@ public function model()
...
+    {
+        $user->tastes()->detach($taste);
+    }
+
+    public function setTastes(User $user, Collection $tastes): void
+    {
+        $user->tastes()->sync($tastes);
+    }
+
+    public function getTastes(User $user): Collection
```
Comment (nemoipaha) [inline app/Repositories/User/UserRepository.php]: Better to have this method in TasteRepository `findByUser(int $userId)`.

File: app/Repositories/UserTaste/TasteRepository.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+namespace Hedonist\Repositories\UserTaste;
+
+use Hedonist\Entities\User\Taste;
+use Illuminate\Support\Collection;
+use Prettus\Repository\Contracts\CriteriaInterface;
+use Prettus\Repository\Eloquent\BaseRepository;
+
+class TasteRepository extends BaseRepository implements TasteRepositoryInterface
+{
+
```
Comment (nemoipaha) [inline app/Repositories/UserTaste/TasteRepository.php]: Empty line and phpdoc remove.

File: app/Repositories/UserTaste/TasteRepository.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Hedonist\Repositories\UserTaste;
```
Comment (nemoipaha) [inline app/Repositories/UserTaste/TasteRepository.php]: You can move under \Repositories\User namespace.

File: app/Repositories/UserTaste/TasteRepositoryInterface.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace Hedonist\Repositories\UserTaste;
```
Comment (nemoipaha) [inline app/Repositories/UserTaste/TasteRepositoryInterface.php]: Namespace fix

File: database/factories/UserTasteFactory.php [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+use Faker\Generator as Faker;
+
+$factory->define(\Hedonist\Entities\User\Taste::class, function (Faker $faker) {
+    return [
+        'name' => $faker->sentence('2')
```
Comment (nemoipaha) [inline database/factories/UserTasteFactory.php]: $faker->word better.

File: database/migrations/2018_08_07_080547_create_tastes_table.php [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('tastes', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('name',45);
```
Comment (nemoipaha) [inline database/migrations/2018_08_07_080547_create_tastes_table.php]: Just string('name').

File: tests/Feature/Repository/UserTastes/UserTastesTest.php [hunk 7]
```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Tests\Feature\Repository\UserTastesTest;
+
```
Comment (nemoipaha) [inline tests/Feature/Repository/UserTastes/UserTastesTest.php]: Empty line.

Comment (klesogor) [reply tests/Feature/Repository/UserTastes/UserTastesTest.php]: I don't really understand what's wrong with this new line? New line after php tag, new line between namespace and use statements

**Pull request merged**